DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/explicit_data_movement/src/explicit_data_movement.cpp

```diff
@@ -0,0 +1,282 @@
+#include <assert.h>
+#include <algorithm>
+#include <chrono>
+#include <iomanip>
+#include <random>
+#include <type_traits>
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+using namespace std::chrono;
+
+// Declare the kernel class name globally to avoid name mangling.
+class ImplicitKernel;
+class ExplicitKernel;
+
+//
+// This version of the kernel demonstrates implicit data movement
+// through SYCL buffers and accessors.
+//
+template<typename T>
+double SubmitImplicitKernel(queue& q, std::vector<T>& in, std::vector<T>& out,
+                            size_t size) {
+  // start the timer
+  auto start = high_resolution_clock::now();
+  
+  {
+    // set up the input and output buffers
+    buffer in_buf(in);
+    buffer out_buf(out);
+
+    // launch the computation kernel
+    auto kernel_event = q.submit([&](handler& h) {
+      accessor in_accessor(in_buf, h, read_only);
```

Comment:
Make kernels be simplest.   Adding _accessor is too verbose.

Reply:
What is the proposed changed here? 'in' would clash with the kernel argument.

Reply:
`in_a` and `out_a`?